kong/plugins/hmac-auth/access.lua

```diff
@@ -129,6 +134,22 @@ local function validate_clock_skew(headers, date_header_name, allowed_clock_skew
   return true
 end
 
+local function validate_request_body(headers, body)
+  local sha_recieved = headers[DIGEST]
+
+  if body == nil then
+    return true
+  elseif sha_recieved == nil then
+    return false
```

Comment:
@vaibhavatul47 here your logic forcing client to send `digest` header. I would prefer logic to be 
```
if validate_request_body and headers[digest] then
  validate digest
end
```